lib/test.d.ts

```diff
@@ -29,40 +35,114 @@ interface TestOptions {
   signal?: AbortSignal;
 }
 
-type TestFn = (t: TestContext) => any | Promise<any>;
+type TestFn = (t: TestContext, done: (result?: any) => void) => any;
 
 export default test;
 
-export function test(name: string, options: TestOptions, fn: TestFn): void;
-export function test(name: string, fn: TestFn): void;
-export function test(options: TestOptions, fn: TestFn): void;
-export function test(fn: TestFn): void;
+/**
+ * @param name The name of the test, which is displayed when reporting test results.
+ *  Default: The name property of fn, or '<anonymous>' if fn does not have a name.
+ * @param options Configuration options for the test.
+ * @param fn The function under test. The first argument to this function is a TestContext object. If the test uses callbacks, the callback function is passed as the second argument.
+ *  Default: A no-op function.
+ * @returns A {@link Promise} resolved with `undefined` once the test completes.
+ */
+export function test(
+  name: string,
+  options: TestOptions,
+  fn: TestFn
```

Comment:
```suggestion
  name?: string,
  options?: TestOptions,
  fn?: TestFn
```

Reply:
👍 made all exported fn params optional in https://github.com/nodejs/node-core-test/pull/42/commits/54dc7d87a87756b99f9b9f8d642f723410d385f5